pumps/sql_test.go

```diff
@@ -119,3 +121,163 @@ func TestSQLWriteDataSharded(t *testing.T) {
 	assert.Len(t, dbRecords, 2)
 
 }
+
+func TestSQLWriteUptimeData(t *testing.T) {
+	pmp := SQLPump{IsUptime: true}
+	cfg := make(map[string]interface{})
+	cfg["type"] = "sqlite"
+	cfg["connection_string"] = "pmp_test.db"
+	cfg["table_sharding"] = false
+	err := pmp.Init(cfg)
+	if err != nil {
+		t.Fatal("SQL Pump couldn't be initialized with err: ", err)
+	}
+	defer func() {
+		os.Remove("pmp_test.db")
+	}()
+
+	keys := make([]interface{}, 3)
+	now := time.Now()
+	nowPlus1 := time.Now().Add(2 * time.Hour)
+
+	encoded, _ := msgpack.Marshal(analytics.UptimeReportData{OrgID: "1", URL: "url1", TimeStamp: now})
```

Comment:
In this test, there is repeating code. Wouldn't it be nicer if we put them in an inner func?

Reply:
@furkansenharputlu do you mean the encoding part?

Reply:
This code is repeating, we can make it modular and create a test func which accepts data and expected length etc.

```encoded, _ = msgpack.Marshal(analytics.UptimeReportData{OrgID: "1", URL: "url1", TimeStamp: nowPlus1})
	keys[0] = string(encoded)
	keys[1] = string(encoded)
	keys[2] = string(encoded)

	pmp.WriteUptimeData(keys)

	dbRecords = []analytics.UptimeReportAggregateSQL{}

	if err := pmp.db.Table(table).Find(&dbRecords).Error; err != nil {
		t.Fatal("Error getting analytics records from SQL")
	}
	assert.Len(t, dbRecords, 4)
```

Reply:
Done @furkansenharputlu